Source/Core/Core/IPC_HLE/WII_IPC_HLE_Device_wfsi.cpp

```diff
@@ -0,0 +1,290 @@
+
+    m_arc_unpacker.AddBytes(decrypted);
+    break;
+  }
+
+  case IOCTL_WFSI_FINALIZE_PROFILE:
+  case IOCTL_WFSI_FINALIZE_CONTENT:
+  {
+    const char* ioctl_name = command == IOCTL_WFSI_FINALIZE_PROFILE ?
+                                 "IOCTL_WFSI_FINALIZE_PROFILE" :
+                                 "IOCTL_WFSI_FINALIZE_CONTENT";
+    INFO_LOG(WII_IPC_HLE, "%s", ioctl_name);
+
+    auto callback = [this](const std::string& filename, const std::vector<u8>& bytes) {
+      INFO_LOG(WII_IPC_HLE, "Extract: %s (%zd bytes)", filename.c_str(), bytes.size());
+
+      std::string path = WFS::NativePath(m_base_extract_path + "/" + filename);
+      File::CreateFullPath(path);
+      File::IOFile f(path, "wb");
+      if (!f)
+      {
+        ERROR_LOG(WII_IPC_HLE, "Could not extract %s to %s", filename.c_str(), path.c_str());
+        return;
+      }
+      f.WriteBytes(bytes.data(), bytes.size());
+    };
+    m_arc_unpacker.Extract(callback);
+
+    // Technically not needed, but let's not keep large buffers in RAM for no
+    // reason if we can avoid it.
+    m_arc_unpacker.Reset();
+    break;
+  }
+
+  case IOCTL_WFSI_DELETE_TITLE:
+    // Bytes 0-4: ??
+    // Bytes 4-8: game id
+    // Bytes 1c-1e: title id?
+    WARN_LOG(WII_IPC_HLE, "IOCTL_WFSI_DELETE_TITLE: unimplemented");
+    break;
+
+  case IOCTL_WFSI_IMPORT_TITLE:
+    WARN_LOG(WII_IPC_HLE, "IOCTL_WFSI_IMPORT_TITLE: unimplemented");
+    break;
+
+  case IOCTL_WFSI_INIT:
+    // Nothing to do.
+    INFO_LOG(WII_IPC_HLE, "IOCTL_WFSI_INIT");
+    break;
+
+  case IOCTL_WFSI_SET_DEVICE_NAME:
+    INFO_LOG(WII_IPC_HLE, "IOCTL_WFSI_SET_DEVICE_NAME");
+    m_device_name = Memory::GetString(buffer_in);
+    break;
+
+  case IOCTL_WFSI_APPLY_TITLE_PROFILE:
+    INFO_LOG(WII_IPC_HLE, "IOCTL_WFSI_APPLY_TITLE_PROFILE");
+
+    m_base_extract_path =
+        StringFromFormat("/vol/%s/_install/%c%c%c%c/content", m_device_name.c_str(),
+                         static_cast<char>(m_title_id >> 24), static_cast<char>(m_title_id >> 16),
+                         static_cast<char>(m_title_id >> 8), static_cast<char>(m_title_id));
+    File::CreateFullPath(WFS::NativePath(m_base_extract_path));
+
+    break;
+
+  default:
+    // TODO(wfs): Should be returning an error. However until we have
+    // everything properly stubbed it's easier to simulate the methods
+    // succeeding.
+    WARN_LOG(WII_IPC_HLE, "%s unimplemented IOCtl(0x%08x, size_in=%08x, size_out=%08x)\n%s\n%s",
+             m_Name.c_str(), command, buffer_in_size, buffer_out_size,
+             HexDump(Memory::GetPointer(buffer_in), buffer_in_size).c_str(),
+             HexDump(Memory::GetPointer(buffer_out), buffer_out_size).c_str());
+    Memory::Memset(buffer_out, 0, buffer_out_size);
+    break;
+  }
+
+  Memory::Write_U32(return_error_code, command_address + 4);
+  return GetDefaultReply();
+}
+
+IPCCommandResult CWII_IPC_HLE_Device_wfsi::IOCtlV(u32 command_address)
+{
+  SIOCtlVBuffer command_buffer(command_address);
```

Comment:
Since we know that ioctlv isn't a valid command and won't be doing anything with it, I don't think we need to parse this IPC request.